test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,312 @@
+                                attrib=attr_A)
+        except PbsManagerError as e:
+            self.assertTrue("Illegal attribute or resource value" in e.msg[0])
+
+        attr_A = {'vnode_pool': 'a'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A)
+        except PbsManagerError as e:
+            self.assertTrue("Illegal attribute or resource value" in e.msg[0])
+
+    def test_two_moms_single_vnode_pool(self):
+        """
+        Same vnode_pool for two moms shall result in one mom being the
+        inventory mom and the other the non-inventory mom.
+        The inventory mom goes down (e.g. killed).
+        Compute nodes remain up even when the inventory mom is killed,
+        since another mom is reporting them.
+        Check that a new inventory mom is listed in the log.
+        Bring up killed mom.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {"log_events": -1})
+
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
+                False, msg="Provide at least 2 moms while invoking test")
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr = {'vnode_pool': '1'}
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr)
+
+        rc = self.server.log_match("Mom %s added to vnode_pool %s" %
+                                   (self.hostB, '1'), max_attempts=5,
+                                   starttime=start_time)
+        self.assertTrue(rc)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        self.assertEqual(found_in_momA + found_in_momB,
+                         1, msg="an inventory mom not chosen correctly")
+
+        # Only one mom is inventory mom
+        if (found_in_momA == 0):
+            inv_mom = self.momA
+            noninv_mom = self.momB
+        else:
+            inv_mom = self.momB
+            noninv_mom = self.momA
+
+        self.logger.info("Inventory mom is %s." % inv_mom.shortname)
+        self.logger.info("Non-inventory mom is %s." %
+                         noninv_mom.shortname)
+
+        start_time = int(time.time())
+
+        # Kill inventory mom
+        inv_mom.stop(sig='-KILL')
```

Comment:
Yes, that's implementation part. But by design signal() is for sending signal to mom (which you are doing in test case) and stop() is for stopping mom. I would still suggest to use .signal() to send KILL command, rest up to you.